yt_dlp/extractor/bluesky.py

```diff
@@ -0,0 +1,379 @@
+from .common import InfoExtractor
+from ..utils import (
+    format_field,
+    int_or_none,
+    mimetype2ext,
+    parse_iso8601,
+    truncate_string,
+    update_url_query,
+    url_or_none,
+    variadic,
+)
+from ..utils.traversal import traverse_obj
+
+
+class BlueskyIE(InfoExtractor):
+    _VALID_URL = [
+        r'https?://(?:www\.)?(?:bsky\.app|main\.bsky\.dev)/profile/(?P<handle>[^/]+)/post/(?P<id>\w+)',
```

Comment:
can we be more strict with handle matching? (maybe the `-` is even unnecessary, i don't see it in any tests)
```suggestion
        r'https?://(?:www\.)?(?:bsky\.app|main\.bsky\.dev)/profile/(?P<handle>[\w.:-]+)/post/(?P<id>\w+)',
```

or if not, then let's at least do this:
```suggestion
        r'https?://(?:www\.)?(?:bsky\.app|main\.bsky\.dev)/profile/(?P<handle>[^/?#]+)/post/(?P<id>\w+)',
```

Reply:
Between the [handle](https://atproto.com/specs/handle) and [DID](https://atproto.com/specs/did) specs, valid characters are letters, digits, periods, underscores, colons, percent signs, and hyphens, so I'm thinking `(?P<handle>[\w.:%-]+)`.

Reply:
edited my suggestions per your feedback